pkg/crud/action.go

```diff
@@ -27,13 +26,22 @@ func CreateAction(object *Action) (*Action, *AppError) {
 	if object.Source.ID != "" {
 		manager.Actions.Find(bson.M{"source.id": object.Source.ID}).One(&dbEntity)
 		if dbEntity.ID != "" {
-			message := fmt.Sprintf("%s exists with source [%s]\n", reflect.TypeOf(object).Name(), object.Source.ID)
+			message := fmt.Sprintf("Action exists with source [%s]\n", object.Source.ID)
 			return nil, &AppError{nil, message, http.StatusInternalServerError}
 		}
 	}
 
+	//do not allow duplicate action from this user
+	user := findUser(object, manager)
+	manager.Actions.Find(bson.M{"user_id": user.ID, "target": object.Target, "type": object.Type}).One(&dbEntity)
```

Comment:
I believe this if statement needs to check against TargetId as well.  

The way I read this it only checks against the target type.  This would only allow a user to have one like on any comment, for example.